### PR TITLE
Implement BeatmapListingSortTabControl component

### PR DESCRIPTION
**Pull request by EVAST9919**
Screenshot_1

### CODE REVIEW

File: osu.Game.Tests/Visual/UserInterface/TestSceneBeatmapListingSort.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            control.Current.BindValueChanged(_ => updateBindablesVisual(), true);
+        }
+
+        private void updateBindablesVisual()
+        {
+            placeholder.Clear();
+
+            placeholder.Add(new OsuSpriteText { Text = $"Current: {control.Current.Value}" });
+            placeholder.Add(new OsuSpriteText { Text = $"Sort direction: {control.SortDirection.Value}" });
+        }
```
Comment (bdach) [inline osu.Game.Tests/Visual/UserInterface/TestSceneBeatmapListingSort.cs]: It's test code, so not too relevant, but a bit staggering nonetheless - you could as well have added the two sprite texts to the hierarchy in the constructor, assigned them to fields and only changed their `.Text` values in the two `BindValueChanged` callbacks.

File: osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        private class BeatmapSortTabControl : SortTabControl
+        {
+            public readonly Bindable<SortDirection> SortDirection = new Bindable<SortDirection>();
+
+            protected override TabItem<BeatmapSortCriteria> CreateTabItem(BeatmapSortCriteria value) => new BeatmapSortTabItem(value)
+            {
+                SortDirection = { BindTarget = SortDirection }
+            };
+
+            private class BeatmapSortTabItem : SortTabItem
```
Comment (bdach) [inline osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs]: Getting a little too nested - can this not be side-by-side with the tab control inside `BeatmapListingSortTabControl`? They're both private anyways.

File: osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                {
+                }
+
+                protected override TabButton CreateTabButton(BeatmapSortCriteria value) => new BeatmapTabButton(value)
+                {
+                    Active = { BindTarget = Active },
+                    SortDirection = { BindTarget = SortDirection }
+                };
+
+                private class BeatmapTabButton : TabButton
```
Comment (bdach) [inline osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs]: Same thing with the nesting here, this doesn't need to be encapsulated this much.

File: osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                    }
+
+                    protected override void LoadComplete()
+                    {
+                        base.LoadComplete();
+
+                        SortDirection.BindValueChanged(direction =>
+                        {
+                            icon.Icon = direction.NewValue == Overlays.SortDirection.Ascending ? FontAwesome.Solid.CaretUp : FontAwesome.Solid.CaretDown;
+                            icon.Margin = direction.NewValue == Overlays.SortDirection.Ascending ? new MarginPadding { Top = 1 } : new MarginPadding { Top = 2 };
```
Comment (bdach) [inline osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs]: This shouldn't be required.

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs (hunk 4 above)
Comment (EVAST9919) [reply osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs]: Are you talking about margin?

Comment (bdach) [reply osu.Game/Overlays/BeatmapListing/BeatmapListingSortTabControl.cs]: Yes, looks fine with it removed.

**Review by bdach — APPROVED** [review]
Looks ok for now.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**